bunsen/bunsen-core/src/main/java/com/cerner/bunsen/definitions/HapiCompositeConverter.java

```diff
@@ -164,15 +164,16 @@ public Object fromHapi(Object input) {
 
     if (composite instanceof IAnyResource) {
 
-      // Id element
       StructureField<HapiConverter<T>> schemaEntry = schemaIterator.next();
-      values[0] = schemaEntry.result().fromHapi(((IAnyResource) composite).getIdElement());
-      valueIndex++;
+      if (schemaEntry.fieldName().equals("id")) {
```

Comment:
So I am thinking about what happens in the reverse direction, i.e., in `toHapiConverter` below. IIUC, with this change, if we convert back an Avro record to HAPI, we create HAPI objects without `id` element, correct? I am not sure if this is okay or if HAPI makes any assertions anywhere that each `Element` should have an id (this is just a question and to make sure we are aware of this and it is okay).

Reply:
hmm, interesting. I am little curious to know the scenarios where we are putting the same avro files back to hapi server. If the requirement is to write to hapi server, then wouldn't it be easier to write Hapi objects directly and skip converting them to avro and avro to hapi?

Reply:
We actually do not rely on the reverse direction ourselves. I am thinking more about Bunsen as a library which is capable of converting Avro records back to HAPI. So for example the scenario can be reading files written by the pipeline before, create HAPI objects out of them and send them to another FHIR server. I _think_ this should still work since `id` is an optional field in [FHIR resources](https://hl7.org/fhir/resource.html).

Reply:
Thanks for confirming this, resolving this conversation then.